lua/diffview/vcs/adapters/hg/init.lua

```diff
@@ -1,18 +1,1141 @@
+    on_exit = on_exit,
+  })
+  local numstat_job = Job:new({
+    command = self:bin(),
+    args = utils.vec_join(self:args(), "diff", "--stat", args),
+    cwd = self.ctx.toplevel,
+    on_exit = on_exit,
+  })
+
+  namestat_job:start()
+  mergestate_job:start()
+  numstat_job:start()
+  latch:await()
+  local out_status
+  if
+    not (#namestat_job:result() == 0 and #numstat_job:result() == 0)
+    and not (#namestat_job:result() == #numstat_job:result() - 1)
+  then
+    out_status =
+      vcs_utils.ensure_output(2, { namestat_job, numstat_job }, "HgAdapter>tracked_files()")
+  end
+
+  if out_status == JobStatus.ERROR or not (namestat_job.code == 0 and numstat_job.code == 0 and mergestate_job.code == 0) then
+    callback(utils.vec_join(namestat_job:stderr_result(), numstat_job:stderr_result(), mergestate_job:stderr_result()), nil)
+    return
+  end
+
+  local numstat_out = numstat_job:result()
+  local namestat_out = namestat_job:result()
+  local mergestate_out = mergestate_job:result()
+
+  local data = {}
+  local conflict_map = {}
+  local file_info = {}
+
+  -- Last line in numstat is a summary and should not be used
+  table.remove(numstat_out, -1)
+
+  for i, s in ipairs(namestat_out) do
+    local status = s:sub(1, 1):gsub("%s", " ")
+    local name = vim.trim(s:match("[%a%s]%s*(.*)"))
+
+    local stats = {}
+    local changes, diffstats = numstat_out[i]:match(".*|%s+(%d+)%s+([+-]+)")
+    if changes and diffstats then
+      local _, adds = diffstats:gsub("+", "")
+
+      stats = {
+        additions = tonumber(adds),
+        deletions = tonumber(changes) - tonumber(adds),
+      }
+    end
+
+    if not (kind == "staged") then
+      file_info[name] = {
+        status = status,
+        name = name,
+        stats = stats,
+      }
+    end
+  end
+
+  local mergestate = vim.json.decode(table.concat(mergestate_out, ''))
+  for _, file in ipairs(mergestate[1].files) do
+    local base = nil
+    for _, extra in ipairs(file.extras) do
+      if extra.key == 'ancestorlinknode' then
+        base = extra.value
+      end
+    end
+    if file.state == 'u' then
+      file_info[file.path].status = 'U'
+      file_info[file.path].base = base
+      if file.other_path ~= file.path then
+        file_info[file.path].oldname = file.other_path
+      end
+      conflict_map[file.path] = file_info[file.path]
+    end
+  end
+
+  local nodes = {}
+  for _, commit in ipairs(mergestate[1].commits) do
+    nodes[commit.name] = commit.node
+  end
+
+  for _, f in pairs(file_info) do
+    if f.status ~= "U" then
+      table.insert(data, f)
+    end
+  end
+
+  if kind == "working" and next(conflict_map) then
+    for _, v in pairs(conflict_map) do
+      table.insert(conflicts, FileEntry.with_layout(opt.merge_layout, {
+        adapter = self,
+        path = v.name,
+        oldpath = v.oldname,
+        status = "U",
+        kind = "conflicting",
+        revs = {
+          a = self.Rev(RevType.COMMIT, nodes['local']),
```

Comment:
```suggestion
          a = self.Rev(RevType.COMMIT, nodes.local),
```

Reply:
`local` is a keyword, so it doesn't allow it as a name :)

Reply:
Ah, right. My bad 😅